Userland/Libraries/LibWeb/EventTiming/PerformanceEventTiming.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2024, Noah Bright <noah.bright.1@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/Bindings/PerformanceEventTimingPrototype.h>
+#include <LibWeb/DOM/Event.h>
+#include <LibWeb/EventTiming/PerformanceEventTiming.h>
+#include <LibWeb/PerformanceTimeline/EntryTypes.h>
+
+namespace Web::EventTiming {
+
+JS_DEFINE_ALLOCATOR(PerformanceEventTiming);
+
+// https://www.w3.org/TR/event-timing/#sec-init-event-timing
+PerformanceEventTiming::PerformanceEventTiming(JS::Realm& realm, String const& name, HighResolutionTime::DOMHighResTimeStamp start_time, HighResolutionTime::DOMHighResTimeStamp duration,
+    DOM::Event event, HighResolutionTime::DOMHighResTimeStamp processing_start, unsigned long long interaction_id)
```

Comment:
Wait, how are we taking a `DOM::Event` by value here? It inherits from `JS::Cell` which is noncopyable. What's going on?

Reply:
I can't explain why this compiles, but I can file an issue and change this to a `DOM:: Event const&`, which was an oversight on my part anyway.

Reply:
Update, I do get the diagnostic `1. The parameter 'event' is copied for each invocation but only used as a const reference; consider making it a const reference [performance-unnecessary-value-param]` from clangd, if that's helpful